cmd/agent/api/listener.go

```diff
@@ -22,10 +23,16 @@ func getIPCAddressPort() (string, error) {
 }
 
 // getListener returns a listening connection
-func getListener() (net.Listener, error) {
-	address, err := getIPCAddressPort()
-	if err != nil {
-		return nil, err
-	}
+func getListener(address string) (net.Listener, error) {
 	return net.Listen("tcp", address)
 }
+
+// returns the host and host:port of the IPC server, and whether it is enabled
+func getIPCServerAddressPort() (string, string, bool) {
+	ipcServerHost := config.Datadog.GetString("agent_ipc_host")
+	ipcServerPort := config.Datadog.GetInt("agent_ipc_port")
+	ipcServerHostPort := net.JoinHostPort(ipcServerHost, strconv.Itoa(ipcServerPort))
+	ipcServerEnabled := ipcServerPort != 0
```

Comment:
non-blocking nit but not sure if we want to short-circuit here:
```suggestion
	ipcServerPort := config.Datadog.GetInt("agent_ipc_port")
    if ipcServerPort <= 0 {
        return "", "", false
    }
    
	ipcServerHost := config.Datadog.GetString("agent_ipc_host")
	ipcServerHostPort := net.JoinHostPort(ipcServerHost, strconv.Itoa(ipcServerPort))
	
	return ipcServerHost, ipcServerHostPort, true
```

Reply:
That's right, fixed it